src/extensions/download_handlers/plugin_examples/microsoft_delta_download_handler/source_update_cache/src/source_update_cache_utils.c

```diff
@@ -149,7 +150,8 @@ ADUC_Result ADUC_SourceUpdateCacheUtils_MoveToUpdateCache(
 {
     ADUC_Result result = { .ResultCode = ADUC_Result_Failure };
     int res = -1;
-    ADUC_FileEntity* fileEntity = NULL;
+    ADUC_FileEntity fileEntity;
+    memset(&fileEntity, 0, sizeof(fileEntity));
```

Comment:
note that in cases like this, you DO want to init fileEntity, as you're unconditionally calling ADUC_FileEntity_Uninit in done, so maybe it's best to always zero it out.

Reply:
agreed. as long as it's zero'ed out then the uninit will noop.